doc/mgmt/gnmi/SONiC_GNOI_Server_Interface_Design.md

```diff
@@ -0,0 +1,632 @@
+
+# Definition/Abbreviation
+| **Term** | **Meaning**                |
+| -------- | -------------------------- |
+| API      | Application Programming Interface |
+| CLI      | Command Line Interface |
+| gRPC     | A modern open-source high performance Remote Procedure Call (RPC) framework that can run in any environment |
+| gNMI     | gRPC Network Management Interface, used to retrieve or manipulate the state of a device via telemetry or configuration data |
+| gNOI     | gRPC Network Operations Interface |
+| NE       | Network Engineers |
+
+# Background
+Software for Open Networking in the Cloud (SONiC) is an open source network operating system (NOS) based on Linux that runs on switches from multiple vendors and ASICs. SONiC offers a full suite of network functionality, like BGP and RDMA, that has been production-hardened in the data centers of some of the largest cloud service providers. 
+
+Currently, most SONiC operation and configuration management requires directly using CLI on the device. This document proposes the addition of gNOI API to SONiC to modernize SONiC operation management. 
+
+GNOI (gRPC network operations interface) is supported through OpenConfig. All gNOI services and documentation are available on the [OpenConfig page](https://github.com/openconfig/gnoi). OpenConfig defines and implements a common, vendor-independent software layer for managing network devices.
+
+# Scope
+This document describes the high level design of SONiC gNOI Server Interface, as well as the gNOI APIs necessary for SONiC operation management. 
+
+GNOI support comes as a natural extension to preexisting gNMI support. GNOI RPCs are exposed on the same server/port as the gNMI server. For more information regarding gNMI, please refer to the [gNMI HLD](https://github.com/ganglyu/SONiC/blob/012afe049a707da87ac258c8aca5c501172d0f33/doc/mgmt/gnmi/SONiC_GNMI_Server_Interface_Design.md).
+
+## 1 Project Goal Summary
+Network engineers (NE) currently rely on command line interfaces (CLI) to operate SONiC devices. We plan to replace CLI by GNOI API, which will help minimize network engineers’ manual touches on SONiC devices.
+
+We plan to use GNOI API to replace the most common NE CLI used for SONiC operation management. These APIs can sit behind a website frontend that NEs interact with directly, which enables NEs to execute common tasks with a simple webpage interaction. This saves time and reduces errors.
+
+## 2 Requirements
+
+Support for commonly used SONiC operation management CLI
+* sudo sonic_installer install
+* sudo reboot
+* sudo systemctl restart <service>
+
+Support for commonly used Linux system management CLI
+* mv
+* cp
+* rm
+
+Support for other commonly used SONiC operations
+* cert installation
+* firmware upgrade
+
+## 3 Architecture Design
+
+All the introduced features are part of the sonic-telemetry package installed in sonic-telemetry container.
```

Comment:
Should be sonic-gnmi package and gnmi container

Reply:
Fixed